.github/workflows/workflow.yml

```diff
@@ -183,256 +153,175 @@ jobs:
         ssh linkerd-docker docker version
         echo ::set-env name=DOCKER_HOST::ssh://linkerd-docker
     - name: Build docker images
+      env:
+        DOCKER_TRACE: 1
       run: |
         export PATH="`pwd`/bin:$PATH"
-        DOCKER_TRACE=1 bin/docker-build
-    - name: Create artifact with CLI and image archives
+        bin/docker-build
+    - name: Create artifact with CLI and image archives (Forked repositories)
       if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork
+      env:
+        ARCHIVES: /home/runner/archives
       run: |
-        mkdir -p $IMAGE_ARCHIVES_DIR
-        cp target/cli/linux/linkerd $IMAGE_ARCHIVES_DIR/
-        TAG="$(CI_FORCE_CLEAN=1 bin/root-tag)"
-        for img in controller grafana proxy web ; do
-          docker save "gcr.io/linkerd-io/$img:$TAG" > $IMAGE_ARCHIVES_DIR/$img.tar
+        mkdir -p $ARCHIVES
+
+        for image in proxy controller web cni-plugin debug cli-bin grafana; do
+          docker save "$DOCKER_REGISTRY/$image:$TAG" > $ARCHIVES/$image.tar &
+          pids[${i}]=$!
+          i=$((i+1))
+        done
+
+        # Wait for `docker save` background processes to complete.
+        for pid in ${pids[*]}; do
+          wait $pid
```

Comment:
I think you can just do one single `wait` with no args and it'll wait for all the children jobs to finish. This also avoids having to track their pids.
BUT, we avoid the option to track individual failures for each job. If we hadn't spawned background jobs, the script would fail at the first failure (since this runs using `bash -e` according to the docs). Spawning background jobs disables that fail-fast behavior (according to my tests), so we'd have to track the exit status for each job and fail whenever one of them is > 0.
We could do that by iterating through the `jobs -f` output and inside the loop do `wait -$pid || exit 1` or something like that.

Reply:
I think something like this will work:

```
for image in proxy controller web cni-plugin debug cli-bin grafana; do
  docker save "$DOCKER_REGISTRY/$image:$TAG" > $ARCHIVES/$image.tar || tee save_fail &
done

# Wait for `docker save` background processes to complete.
wait < <(jobs -p)
test -f save_fail && exit 1
```

1. Each background job will create a `save_fail` file if the `docker save ..` command fails.
2. All the background jobs are waited for
3. Test for the presence of `save_fail` fail and exit if it is found.

I can test this a little more, but good find here!